zilencer/management/commands/rundjangoserver.py

```diff
@@ -1,46 +1,25 @@
 from datetime import datetime
-from typing import Protocol
 
 from django.core.management.commands.runserver import Command as DjangoCommand
 
+# Might fail to suppress the date line around midnight, but, whatever.
+date_prefix = datetime.now().strftime("%B %d, %Y - ")
 
-class Writable(Protocol):
-    def write(self, s: str) -> None:
-        ...
 
-
-class FakeStdout:
-    """
-    Filter stdout from Django's runserver, to declutter run-dev.py
-    startup output.
-    """
-
-    def __init__(self, stdout: Writable) -> None:
-        self.stdout = stdout
-        # Might fail to suppress the date line around midnight, but, whatever.
-        self.date_prefix = datetime.now().strftime("%B %d, %Y - ")
-
-    def write(self, s: str) -> None:
-        if (
-            s == "Performing system checks...\n\n"
-            or s.startswith("System check identified no issues")
-            or s.startswith(self.date_prefix)
-        ):
-            pass
-        elif "Quit the server with " in s:
-            self.stdout.write(
-                "Django process (re)started. " + s[s.index("Quit the server with ") :]
-            )
-        else:
-            self.stdout.write(s)
+def output_styler(message: str) -> str:
+    if (
+        message == "Performing system checks...\n\n"
+        or message.startswith("System check identified no issues")
+        or message.startswith(date_prefix)
+    ):
+        return ""
+    elif "Quit the server with " in message:
+        return "Django process (re)started. " + message[message.index("Quit the server with ") :]
+    else:
+        return message
 
 
 class Command(DjangoCommand):
-    stdout: Writable
-
     def inner_run(self, *args: object, **options: object) -> None:
-        self.stdout = FakeStdout(self.stdout)
-        try:
-            super().inner_run(*args, **options)
-        finally:
-            self.stdout = self.stdout.stdout
+        self.stdout.style_func = output_styler
```

Comment:
Theoretically, Django uses the original `style_func` to color the output. I haven’t found anything that it actually colors, but in case it colors more in the future, we should make the new `style_func` wrap the old one.